pilot/pkg/xds/endpoint_builder.go

```diff
@@ -230,9 +231,68 @@ func (b *EndpointBuilder) populateFailoverPriorityLabels() {
 // build LocalityLbEndpoints for a cluster from existing EndpointShards.
 func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 	shards *model.EndpointShards,
+	failoverShards *model.EndpointShards,
 	svcPort *model.Port,
+	failOverPort uint32,
 ) []*LocalityEndpoints {
 	localityEpMap := make(map[string]*LocalityEndpoints)
+	locEps := make([]*LocalityEndpoints, 0, len(localityEpMap))
+
+	if shards != nil {
```

Comment:
We can't just swap out endpoints for a cluster. Endpoints are tightly bound to Clusters. If we failover we would now have random endpoints from some other service, but still the metadata, SAN matching, etc from the original Service. Additionally ,they may be entirely different types (DNS vs STATIC, etc)